packages/pwa/app/utils/locale.js

```diff
@@ -6,44 +6,46 @@
  */
 
 import PropTypes from 'prop-types'
-import {SUPPORTED_LOCALES, DEFAULT_LOCALE} from '../constants'
-
-const supportedLocales = SUPPORTED_LOCALES.map((locale) => locale.id)
 
 /**
  * @returns {string[]} short codes of all the app's supported locales
  */
-export const getSupportedLocalesIds = () => supportedLocales
+export const getSupportedLocalesIds = (supportedLocales = []) =>
+    supportedLocales.map((locale) => locale.id)
 
 /**
  * Dynamically import the translations/messages for a given locale
  * @private
- * @param {string} locale - The locale code
- * @returns {Promise<Object>} The messages (compiled in AST format) in the given locale. If locale is not found, returns the default locale's messages.
+ * @param {string} targetLocale
+ * @param {string} defaultLocale
+ * @param {string[]} supportedLocales
+ * @returns {Promise<Object>} The messages (compiled in AST format) in the given locale. If locale is not found, returns the default locale's messages. If the translation file is not found, return the default messages instead.
  */
-export const loadLocaleData = async (locale) => {
+export const loadLocaleData = async (targetLocale, defaultLocale, supportedLocales) => {
```

Comment:
So I'm assuming that the signature of this function was changed because we have multiple sites with multiple supported locales, each with their own default locale. 

But I feel like it's not entirely correct to pass in, the target, default and supported locals and having the function verify that the target is in the supported list. That is kind of like going to a bank teller and saying "I'd like to withdraw 50k, and don't worry I have 100K in the bank" the validation is that you have enough money is irrelevant for the teller to check.

So a better function signature might be something like `loadLocaleData(locale, fallback)`. Then you don't have to use this useless check.

Reply:
Thanks for the suggestion, it seems that this function can be even simpler by just passing the locale, we don't need the fallback as it has been handled by `whichLocaleToLoad` which defines the targetLocale before passing it to`loadLocaleData`